internal/controller/monovertexrollout/monovertexrollout_progressive.go

```diff
@@ -45,55 +42,56 @@ func (r *MonoVertexRolloutReconciler) CreateUpgradingChildDefinition(ctx context
 
 // AssessUpgradingChild makes an assessment of the upgrading child to determine if it was successful, failed, or still not known
 // This implements a function of the progressiveController interface
-func (r *MonoVertexRolloutReconciler) AssessUpgradingChild(ctx context.Context, rolloutObject progressive.ProgressiveRolloutObject, existingUpgradingChildDef *unstructured.Unstructured) (apiv1.AssessmentResult, string, error) {
+func (r *MonoVertexRolloutReconciler) AssessUpgradingChild(
+	ctx context.Context,
+	rolloutObject progressive.ProgressiveRolloutObject,
+	existingUpgradingChildDef *unstructured.Unstructured,
+	assessmentSchedule config.AssessmentSchedule) (apiv1.AssessmentResult, string, error) {
+	numaLogger := logger.FromContext(ctx)
+
 	mvtxRollout := rolloutObject.(*apiv1.MonoVertexRollout)
-	analysis := mvtxRollout.GetAnalysis()
-	// only check for and create AnalysisRuns if templates are specified
-	if len(analysis.Templates) > 0 {
-		analysisRun := &argorolloutsv1.AnalysisRun{}
-		// check if analysisRun has already been created
-		if err := r.client.Get(ctx, client.ObjectKey{Name: existingUpgradingChildDef.GetName(), Namespace: existingUpgradingChildDef.GetNamespace()}, analysisRun); err != nil {
-			if apierrors.IsNotFound(err) {
-				// analysisRun is created the first time the upgrading child is assessed
-				ownerRef := *metav1.NewControllerRef(&metav1.ObjectMeta{Name: existingUpgradingChildDef.GetName(), Namespace: existingUpgradingChildDef.GetNamespace(), UID: existingUpgradingChildDef.GetUID()}, numaflowv1.MonoVertexGroupVersionKind)
-				promotedChildStatus := rolloutObject.GetPromotedChildStatus()
-				var promotedChildName string
-				if promotedChildStatus != nil {
-					promotedChildName = promotedChildStatus.Name
-				}
-				err := progressive.CreateAnalysisRun(ctx, analysis, existingUpgradingChildDef, ownerRef, r.client, promotedChildName)
+
+	childStatus := mvtxRollout.GetUpgradingChildStatus()
+
+	assessment, reasonFailure, err := progressive.PerformResourceHealthCheckForPipelineType(ctx, existingUpgradingChildDef, progressive.AreVertexReplicasReady)
+	if err != nil {
+		return assessment, reasonFailure, err
+	}
+	if assessment == apiv1.AssessmentResultFailure {
+		// set AssessmentEndTime to now and return failure
+		assessmentEndTime := metav1.NewTime(time.Now())
+		childStatus.BasicAssessmentEndTime = &assessmentEndTime
+		mvtxRollout.SetUpgradingChildStatus(childStatus)
+		return assessment, reasonFailure, nil
+	}
+	if assessment == apiv1.AssessmentResultSuccess {
+		// has AssessmentEndTime been set? if not, set it - now we can start our interval
+		if !childStatus.IsAssessmentEndTimeSet() {
+			assessmentEndTime := metav1.NewTime(time.Now().Add(assessmentSchedule.Period))
+			childStatus.BasicAssessmentEndTime = &assessmentEndTime
+			numaLogger.WithValues("childStatus", *childStatus).Debug("set upgrading child AssessmentEndTime")
+			mvtxRollout.SetUpgradingChildStatus(childStatus)
```

Comment:
This logic for determining the Assessment End Time used to be in `progressive.go`, but I needed to move it to here and to `pipelinerollout_progressive.go` because these `AssessUpgradingChild()` functions are now being called all the way from the beginning of the resource health check window to when the AnalysisRun is complete (if there is an AnalysisRun).